rasa/core/training/loading.py

```diff
@@ -0,0 +1,146 @@
+import logging
+import os
+from pathlib import Path, PurePath
+from typing import Text, Optional, Dict, List, Union, TYPE_CHECKING
+
+import rasa.utils.io as io_utils
+from rasa.core.domain import Domain
+from rasa.core.interpreter import NaturalLanguageInterpreter, RegexInterpreter
+from rasa.data import YAML_FILE_EXTENSIONS, MARKDOWN_FILE_EXTENSION
+
+from rasa.core.training.structures import StoryStep
+from rasa.core.training.story_reader.story_reader import StoryReader
+from rasa.core.training.story_reader.markdown_story_reader import MarkdownStoryReader
+from rasa.core.training.story_reader.yaml_story_reader import YAMLStoryReader
+
+logger = logging.getLogger(__name__)
+
+
+def _get_reader(
+    filename: Text,
+    domain: Domain,
+    interpreter: NaturalLanguageInterpreter = RegexInterpreter(),
+    template_variables: Optional[Dict] = None,
+    use_e2e: bool = False,
+) -> StoryReader:
+
+    reader = None
+
+    if filename.endswith(MARKDOWN_FILE_EXTENSION):
+        reader = MarkdownStoryReader(
+            interpreter, domain, template_variables, use_e2e, filename
+        )
+    elif PurePath(filename).suffix in YAML_FILE_EXTENSIONS:
```

Comment:
that doesn't work? Why do we need `PurePath`?
```suggestion
    elif Path(filename).suffix in YAML_FILE_EXTENSIONS:
```

Reply:
`Path` is a child of `PurePath`, and the `pathlib` docs mention `PurePath(filename).suffix` as an example, I simply followed it.
But since you've found that confusing I can change that.